Google.PowerShell.IntegrationTests/BigQuery/BigQuery.GcbqDataset.Tests.ps1

```diff
@@ -0,0 +1,93 @@
+		$data | Should Not BeNullOrEmpty
+		$data.DatasetReference.DatasetId | Should Be "test_data_id1"
+		$data.FriendlyName | Should Be "Testdata"
+		$data.Description | Should Be "Some interesting data!"
+		$data.DefaultTableExpirationMs | Should Be 86400000
+	}
+
+	It "should accept a dataset object from pipeline" {
+		$data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+		$data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+		$data.DatasetReference.DatasetId = "test_data_id2"
+		$data.DatasetReference.ProjectId = $project
+		$data.FriendlyName = "PipeTest"
+		$data.Description = "Some cool data coming hot off the pipeline!"
+		$data | New-GcbqDataset -outvariable newdata
+		$newdata | Should Not BeNullOrEmpty
+		$newdata.DatasetReference.DatasetId | Should Be "test_data_id2"
+		$newdata.FriendlyName | Should Be "PipeTest"
+	}
+
+	It "should accept a dataset object from a command line argument" {
+		$data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+		$data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+		$data.DatasetReference.DatasetId = "test_data_id3"
+		$data.DatasetReference.ProjectId = $project
+		$data.FriendlyName = "PipeTest"
+		$data.Description = "Some cool data coming hot off the pipeline!"
+		$newdata = New-GcbqDataset -Dataset $data
+		$newdata | Should Not BeNullOrEmpty
+		$newdata.DatasetReference.DatasetId | Should Be "test_data_id3"
+		$newdata.FriendlyName | Should Be "PipeTest"
+	}
+
+	It "should not work with no arguments (meaning no datasetId)" {
+		{ New-GcbqDataset } | Should Throw 400
+	}
+
+	It "should reject datasets with malformed ids" {
+		{ New-GcbqDataset "test-?ata-4" } | Should Throw 400
+	}
+
+	It "should reject datasets with empty ids" {
+		{ New-GcbqDataset "" } | Should Throw "Cannot validate argument on parameter 'DatasetId'"
+	}
+
+	It "should reject datasets that do not have unique ids" {
+		$data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+		$data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+		$data.DatasetReference.DatasetId = "test_data_id6"
+		$data2 = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+		$data2.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+		$data2.DatasetReference.DatasetId = "test_data_id6"
+		$data | New-GcbqDataset
+		{ $data2 | New-GcbqDataset } | Should Throw 409
+	}
+
+	It "should use the -Project tag correctly" {
+		$data = New-GcbqDataset -Project $project "test_data_id7" -Name "Testdata" -Description "Some interesting data!"
+		$data | Should Not BeNullOrEmpty
+		$data.DatasetReference.ProjectId | Should Be $project
+	}
+
+	It "should throw when you try to add to a project that doesnt exist" {
+		{ New-GcbqDataset -Project $nonExistProject "test_data_id8" -Name "Testdata" -Description "Some interesting data!" } | Should Throw 404
```

Comment:
Prefer spaces over tabs. That is to be consistent with every language Google uses but Go. Go uses Tabs over spaces, because "spaces for alignment, tabs for indentation".

"Use the default Code Editor settings (smart indenting, four-character indents, tabs saved as spaces)." 4-spaces to deliniate blocks.

https://msdn.microsoft.com/en-us/library/ff926074.aspx